accounts/views.py

```diff
@@ -143,4 +94,33 @@ def test_token(request):
         json: validation msg
     """
     print(request.user)
-    return Response("passed for {}".format(request.user.username))
\ No newline at end of file
+    return Response("passed for {}".format(request.user.username))
+
+@api_view(['GET'])
+def get_user_by_username(request, username):
+    try:
+        user = User.objects.get(username=username)
+        serializer = UserPublicSerializer(user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    except User.DoesNotExist:
+        return Response({"detail": "User not found"}, status=status.HTTP_404_NOT_FOUND)
+    
+@api_view(['GET'])
+@authentication_classes([SessionAuthentication, TokenAuthentication])
+@permission_classes([IsAuthenticated])
+def get_user_personal_profile(request, username):
```

Comment:
Shouldn't it be done without the username parameter? There would be a check of which user is logged in.

Reply:
I will do smaller pull requests :D .... i just wanted to have prototype done so thats why....this is last PR of that kind....

Regarding to parameter....we use username as PK in DB for user...and its piece of data allready fetched on FE...so thats why username.